app/src/org/commcare/views/widgets/DateWidget.java

```diff
@@ -66,6 +69,8 @@ private DatePicker buildDatePicker(boolean isQuestionWriteable) {
             public void run() {
                 widgetEntryChanged();
                 isRelevancyUpdateScheduled = false;
+                blockingActionsManager.registerActionCompletion(
```

Comment:
In general with these kinds of callback-centric workflows it's pretty dangerous for the way that the callback to be completed to rely on the callback being handled in this kind of manner.